internal/pkg/aws/cloudformation/changeset.go

```diff
@@ -0,0 +1,152 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package cloudformation
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/google/uuid"
+)
+
+const (
+	// The change set name must match the regex [a-zA-Z][-a-zA-Z0-9]*. The generated UUID can start with a number,
+	// by prefixing the uuid with a word we guarantee that we start with a letter.
+	fmtChangeSetName = "ecscli-%s"
+
+	// Status reasons that can occur if the change set execution status is "FAILED".
+	noChangesReason = "NO_CHANGES_REASON"
+	noUpdatesReason = "NO_UPDATES_REASON"
+)
+
+type changeSet struct {
+	name      string
+	stackName string
+	client    changeSetAPI
+}
+
+type changeSetDescription struct {
+	executionStatus string
+	statusReason    string
+	changes         []*cloudformation.Change
+}
+
+func newChangeSet(cfnClient changeSetAPI, stackName string) (*changeSet, error) {
+	id, err := uuid.NewRandom()
+	if err != nil {
+		return nil, fmt.Errorf("generate random id for Change Set: %w", err)
+	}
+
+	return &changeSet{
+		name:      fmt.Sprintf(fmtChangeSetName, id.String()),
+		stackName: stackName,
+
+		client: cfnClient,
+	}, nil
+}
+
+func (cs *changeSet) String() string {
+	return fmt.Sprintf("change set %s for stack %s", cs.name, cs.stackName)
+}
+
+// create creates a Change Set with type CREATE and waits until it's created.
```

Comment:
Well the type is passed in - so I think this comment should prolly be updated :D

Reply:
updated!